### PR TITLE
Hotfix/change in axis

### PR DESCRIPTION
**Pull request by achshar**
Description:
Describe your pull request and attach links to any issues that may be related.

### CODE REVIEW

File: client/js/reports-manager.js [hunk 1]
```diff
@@ -2321,76 +2319,153 @@ class ReportVisualizationLinearOptions extends ReportVisualizationOptions {
...
+			container = this.axisContainer = document.createElement('div'),
 			datalist = [];
 
 		container.classList.add('axis', 'subform');
 
-		for(const [key, column] of this.page.preview.report.columns)
+		for(const [key, column] of this.axes.columns)
 			datalist.push({name: column.name, value: key});
 
+		let usedColumns = {};
```
[the remark this answers is not in the document]
Comment (Ravikant96) [reply client/js/reports-manager.js]: As discussed, if we change the checks below then there we need to assign value to `usedColumns`.

File: client/js/reports-manager.js [hunk 2]
```diff
@@ -2321,76 +2319,150 @@ class ReportVisualizationLinearOptions extends ReportVisualizationOptions {
...
+class Axis {
+
+	constructor(axis, axes) {
+
+		for(const key in axis)
+			this[key] = axis[key]
+
+		this.axes = axes;
+
+		this.axes.columns = this.axes.stage.page.preview.report.columns;
```
Comment (achshar) [inline client/js/reports-manager.js]: Not needed.

File: client/js/reports-manager.js [hunk 3]
```diff
@@ -2321,76 +2319,150 @@ class ReportVisualizationLinearOptions extends ReportVisualizationOptions {
...
 			datalist.push({name: column.name, value: key});
 
+		let usedColumns = [];
+
+		for(const axis of this.axes) {
+
+			if(axis.position == this.position)
+				continue;
+
+			axis.columns = axis.columns ? axis.columns : [];
```
Comment (achshar) [inline client/js/reports-manager.js]: Move to constructor

File: client/js/reports-manager.js [hunk 4]
```diff
@@ -2424,44 +2496,104 @@ class ReportVisualizationLinearOptions extends ReportVisualizationOptions {
...
+
+			let usedColumns = [];
+			const freeColumns = [];
+
+			for(const axis of this.axes)
+				usedColumns = usedColumns.concat(axis.container.multiSelectColumns.value);
+
+			for(const axis of this.axes) {
+				for(const item of axis.container.multiSelectColumns.datalist) {
+					const a = freeColumns.map(c => c.value);
```
Comment (achshar) [inline client/js/reports-manager.js]: a

Comment (achshar) [inline client/js/reports-manager.js]: Merge with the `includes` below.

File: client/js/reports-manager.js [hunk 5]
```diff
@@ -2424,44 +2496,104 @@ class ReportVisualizationLinearOptions extends ReportVisualizationOptions {
...
+			}
+
+			for(const axis of this.axes) {
+
+				if(axis == this)
+					continue;
+
+				const selected = axis.container.multiSelectColumns.value;
+
+				var newDataList = [];
```
Comment (achshar) [inline client/js/reports-manager.js]: var

**Pull request merged**